### DISCUSSION TITLE
Many warnings

### DISCUSSION THREAD

**Discussion by nachoalcazar** [quasarframework/app-extension-qpdfviewer#110] [General]
Hello!!!

I'm testing the extension but when I try to load the pdf I see many warnings:

![image](https://user-images.githubusercontent.com/94051561/147746876-44e481e9-df1f-438f-b05e-6637a77a8991.png)

I think the problem is viewer.properties

Do you know how the component load the file? I thik it should use:

![image](https://user-images.githubusercontent.com/94051561/147747369-bb73a7d5-ac2b-4935-adf9-fa1f529b7d11.png)

In the other hand... is there any slot, event... to show that component is loading?

Greetings

**Comment by hawkeye64** [comment]
Are you sure you loaded the correct one?

Assuming you are using Quasar v2/Vue v3, then this would be the correct wat to install:

```
quasar ext add @quasar/qpdfviewer@next
```

**Reply by nachoalcazar** [reply]
First time I install:

`quasar ext add @quasar/qpdfviewer `


and I has an error: _"Extension(@quasar/qpdfviewer): is not compatible with @quasar/app v3.2.6. Required version: ^1.0.0 || ^2.0.0"_

I uninstall with:

`quasar ext remove @quasar/qpdfviewer`

and now i'm Installing:

`quasar ext add @quasar/qpdfviewer@next`

...but I have all these warnings. I have this structure:

![image](https://user-images.githubusercontent.com/94051561/147765624-1dad7eba-5404-49f5-b1c6-422e4ff79c39.png)

**Reply by hawkeye64** [reply]
Sorry, I have no ideas. When I install on a new project I am not getting any issues.
What's your environment?

**Reply by nachoalcazar** [reply]
If it doesn't sound familiar to you, I'm going to "googlear" it to see where it gets the locale from and how it references these files

**Reply by nachoalcazar** [reply]
In Google Chrome with language "german" works OK:

![image](https://user-images.githubusercontent.com/94051561/147784529-ffde9658-b6e9-4b1f-a706-996c5a793cf2.png)

There is some problem with spanish :-(

**Reply by hawkeye64** [reply]
Look at their Spanish translations compared to German and see if they are similar?
Otherwise, this issue may have to go to their repo.

**Reply by nachoalcazar** [reply]
Yes, not your problem.

The problem was, some browsers, in console:

(this works OK)
![image](https://user-images.githubusercontent.com/94051561/147796033-f8783186-2fc5-4426-8053-5f223cb29d0b.png)

(this NO WORKS)

![image](https://user-images.githubusercontent.com/94051561/147796061-0574b9f8-d232-4b03-88a6-66ff61d418a8.png)

If you copy/paste 
...public\pdfjs\web\locale\es-ES 
to 
...\public\pdfjs\web\locale\es

![image](https://user-images.githubusercontent.com/94051561/147796152-0d16968a-c3e6-4aa6-bb8d-9f7ef5b8d2a6.png)


and insert next lines;

...\public\pdfjs\web\locale\locale.properties
![image](https://user-images.githubusercontent.com/94051561/147796116-ea69c2b6-56f8-48d4-9fe5-361c67e21091.png)

Runs OK

**Reply by hawkeye64** [reply]
Have you written an issue against their repo? They should have 'es' as a minimum. If the browser was 'es-Es' and that file failed to load, then they would grab the start part and load 'es', so it's essential they have the base for all languages. The extension part is not always necessary. For instance, if the browser was 'pt-BR' and they did not have that file, they can grab the front part, 'pt', and load that. It may look a bit different for Brazilians, but they would understand the Portuguese.

**Reply by nachoalcazar** [reply]
mozilla/pdf.js#13689

> The PR above should have fixed this if I look at https://github.com/mozilla/pdf.js/pull/13696/files#diff-51a0dde82a8d1b56568a0375e28fccde02bc154a89b3cb92f34f090b5219b62aR100. Are you sure you're using the latest PDF.js version?

**Reply by hawkeye64** [reply]
That could be it. I'll look into using to the latest one the app-extension and doing a release.

**Reply by nachoalcazar** [reply]
Thanks Jeff

**Reply by nachoalcazar** [reply]
Do you know why later update, the extension continues in this list?

![image](https://user-images.githubusercontent.com/94051561/147879902-7e043883-446c-40b3-b066-bc33fe2b7e23.png)

**Reply by hawkeye64** [reply]
Ah, sorry - you need to reinvoke or add again the app-extension for it to render the new PDFjs. I should have put that into the release notes. I'll do that ASAP. In the mean time:

```
quasar ext add @quasar/qpdfviewer
# or, after updating version
quasar ext invoke @quasar/qpdfviewer
```

**Reply by hawkeye64** [reply]
Apologies I completely forgot about that. Unfortunately, we could never get PDFjs to run from `node_modules` and had to do it this way to work.
